src/nvim/memory.c

```diff
@@ -488,13 +511,13 @@ void time_to_bytes(time_t time_, uint8_t buf[8])
 void free_all_mem(void)
 {
   buf_T       *buf, *nextbuf;
+  static bool entered = false;
 
-  // When we cause a crash here it is caught and Vim tries to exit cleanly.
-  // Don't try freeing everything again.
-  if (entered_free_all_mem) {
```

Comment:
It was added recently, as a merge of Vim's 7.4.1868.  See 945540b7825e2909d17d8b8867330c3ace7c759c.